src/ranged.cpp

```diff
@@ -254,27 +275,33 @@ bool player::handle_gun_damage( item &it )
             }
         }
         // These are the dirtying/fouling mechanics
-        if( dirt < 10000 ) {
-            dirtadder = curammo_effects.count( "BLACKPOWDER" ) * ( 200 - ( firing->blackpowder_tolerance *
-                        2 ) );
-            if( dirtadder < 0 ) {
-                dirtadder = 0;
+        if( !curammo_effects.count( "NON-FOULING" ) && !it.has_flag( "NON-FOULING" ) ) {
+            if( dirt < 10000 ) {
+                dirtadder = curammo_effects.count( "BLACKPOWDER" ) * ( 200 - ( firing->blackpowder_tolerance *
+                            2 ) ); // this is the dirt-increasing number for shots fired with gunpowder-based ammo. Usually dirt level increases by 1, unless it's blackpowder, in which case it increases by a higher number, but there is a reduction for blackpowder resistance of a weapon.
+                if( dirtadder < 0 ) {
+                    dirtadder = 0;
+                }
+                if( curammo_effects.count( "BLACKPOWDER" ) ||
+                    dirt < 7150 ) { // in addition to increasing dirt level faster, regular gunpowder fouling is also capped at 7,150, not 10,000. So firing with regular gunpowder can never make the gun quite as bad as firing it with black gunpowder. At 7,150 the chance to jam is significantly lower (though still significant) than it is at 10,000, the absolute cap.
+                    it.set_var( "dirt", std::min( 10000, dirt + dirtadder + 1 ) );
+                }
             }
-            it.set_var( "dirt", std::min( 10000, dirt + dirtadder + 1 ) );
-        }
-        dirt = it.get_var( "dirt", 0 );
-        dirt_dbl = static_cast<double>( dirt );
-        if( dirt > 0 && !it.faults.count( fault_gun_blackpowder ) ) {
-            it.faults.insert( fault_gun_dirt );
-        }
-        if( dirt > 0 && curammo_effects.count( "BLACKPOWDER" ) ) {
-            it.faults.erase( fault_gun_dirt );
-            it.faults.insert( fault_gun_blackpowder );
+            dirt = it.get_var( "dirt", 0 );
+            dirt_dbl = static_cast<double>( dirt );
+            if( dirt > 0 && !it.faults.count( fault_gun_blackpowder ) ) {
+                it.faults.insert( fault_gun_dirt );
+            }
+            if( dirt > 0 && curammo_effects.count( "BLACKPOWDER" ) ) {
+                it.faults.erase( fault_gun_dirt );
+                it.faults.insert( fault_gun_blackpowder );
+            }
+            // end fouling mechanics
         }
-        // end fouling mechanics
     }
     if( dirt_dbl > 5000 &&
-        x_in_y( dirt_dbl * dirt_dbl * dirt_dbl, 5555555555555 ) ) {
+        x_in_y( dirt_dbl * dirt_dbl * dirt_dbl,
+                5555555555555 ) ) { // chance to damage gun due to high levels of dirt. Very unlikely, especially at lower levels and impossible below 5,000. Lower than the chance of a jam at the same levels.
```

Comment:
Here I also meant what the significance of `5555555555555` was, but this is also useful info.